src/query/src/promql/planner.rs

```diff
@@ -2500,6 +2510,71 @@ impl PromPlanner {
         Ok(scalar_plan)
     }
 
+    /// Create a [SPECIAL_ABSENT_FUNCTION] plan
+    async fn create_absent_plan(
+        &mut self,
+        args: &PromFunctionArgs,
+        query_engine_state: &QueryEngineState,
+    ) -> Result<LogicalPlan> {
+        if args.args.len() != 1 {
+            return FunctionInvalidArgumentSnafu {
+                fn_name: SPECIAL_ABSENT_FUNCTION.to_string(),
+            }
+            .fail();
+        }
+        let input = self
+            .prom_expr_to_plan(&args.args[0], query_engine_state)
+            .await?;
+
+        let time_index_expr = self.create_time_index_column_expr()?;
+        let first_field_expr =
+            self.create_field_column_exprs()?
+                .pop()
+                .with_context(|| ValueNotFoundSnafu {
+                    table: self.ctx.table_name.clone().unwrap_or_default(),
+                })?;
+        let first_value_expr = first_value(first_field_expr, None);
```

Comment:
Do we need to respect the `__filed__` matcher? Looks like it always use the first field here.

Reply:
the field list here is after `__filed__` logic. It's handled when constructing selector